File: src/main/java/com/andredonadon/sorting/Insertion.java
package com.andredonadon.sorting;

public class Insertion {

    public static void insertionSort(int[] arr) {
        int n = arr.length;
        for (int i = 1; i < n; i++) {
            int key = arr[i];
            int j = i - 1;

            // Move os elementos do array que são maiores que key para a frente
            while (j >= 0 && arr[j] > key) {
                arr[j + 1] = arr[j];
                j = j - 1;
            }
            arr[j + 1] = key;
        }
    }

    public static void main(String[] args) {
        int[] arr = {5, 2, 4, 6, 1, 3, 9, 7};

        System.out.println("Array original:");
        printArray(arr);

        insertionSort(arr);

        System.out.println("Array ordenado:");
        printArray(arr);
    }

    public static void printArray(int[] arr) {
        for (int num : arr) {
            System.out.print(num + " ");
        }
        System.out.println();
    }

}

File: kalgorithms/src/main/kotlin/com/andredonadon/kalgorithms/sorting/Insertion.kt
package com.andredonadon.kalgorithms.sorting

fun insertionSort(arr: IntArray) {
    for (i in 1 until arr.size) {
        val key = arr[i]
        var j = i -1

        while (j >= 0 && arr[j] > key) {
            arr[j+1] = arr[j]
            j--
        }

        arr[j+1] = key
    }
}

fun main() {
    val arr = intArrayOf(5, 2, 4, 6, 1, 3, 9, 7)

    println("Array original: ${arr.joinToString(", ")}")

    insertionSort(arr)

    println("Array ordenado: ${arr.joinToString(", ")}")
}